Support  BIGINT Primary Keys

labkey-matthewb opened the pull request:
#### Rationale
Implement support for tables with BIGINT primary keys.  E.g. RowId/ObjectId BIGSERIAL
Because of the large number of shared classes, utilities, services, etc, a lot of code was migrated to used Long.  Code should now assume that an generic integer object key (as opposed to a known table PK), could be a Long.

This PR is aimed at a) Supporting ObjectID BIGSERIAL b) making it possible (but not automatic) to migrate tables that want to use RowId BIGSERIAL.

#### Related Pull Requests
- <!-- list of links to related pull requests (replace this comment) -->

#### Changes
New safer collection classes
  class Int/LongHashMap
  class Int/LongHashSet
explicit Class for key values in Map Objects returned by BaseSelector
  Selector.getValueMap(Class key)
safe "cast" using asInteger or asLong, as opposed to (Integer) and (Long)

<!-- list of standard tasks (remove this comment to enable)
#### Tasks 📍
- [ ] Manual Testing
- [ ] Needs Automation
- [ ] Verify Fix
-->

labkey-matthewb committed:
checkpoint - migrate to using long/Long to represent any integer ID value (rowId, objectId, etc).

labkey-matthewb committed:
checkpoint - migrate to using long/Long to represent any integer ID value (rowId, objectId, etc).

labkey-matthewb committed:
null check

labkey-matthewb committed:
Merge remote-tracking branch 'origin/develop' into fb_long_is_the_new_int

labkey-matthewb committed:
Fix PlateManager

labkey-matthewb committed:
simplify

labkey-matthewb committed:
deleteStatus()

labkey-matthewb committed:
Merge remote-tracking branch 'origin/develop' into fb_long_is_the_new_int

labkey-matthewb committed:
SampleId -> Long

labkey-matthewb committed:
Merge remote-tracking branch 'origin/develop' into fb_long_is_the_new_int

labkey-matthewb committed:
Merge remote-tracking branch 'origin/develop' into fb_long_is_the_new_int

labkey-matthewb committed:
revert

labkey-matthewb committed:
collections -> collections4

labkey-matthewb committed:
fix cast

labkey-matthewb committed:
PlateDesigner

labkey-matthewb committed:
getValueMap() -> getValueMap(Class)

labkey-matthewb committed:
go all in on IntHashMap and LongHashMap

labkey-matthewb committed:
go all in on IntHashMap and LongHashMap

labkey-matthewb committed:
go all in on IntHashMap and LongHashMap

labkey-matthewb committed:
IntArrayList
LongArrayList

labkey-matthewb committed:
Nab Integer->Long

labkey-matthewb committed:
Merge remote-tracking branch 'origin/develop' into fb_long_is_the_new_int

labkey-matthewb committed:
mismatched getter/setter

labkey-matthewb committed:
instanceof Integer
(TODO look for more of these)

labkey-matthewb committed:
Merge remote-tracking branch 'origin/develop' into fb_long_is_the_new_int

labkey-matthewb committed:
mmismatched getter/setter

labkey-matthewb committed:
NPE

labkey-matthewb committed:
RowId BIGINT

labkey-matthewb committed:
StudyPublish Map<Object> -> Map<Long>

labkey-matthewb committed:
Map<Object>

labkey-matthewb committed:
Merge remote-tracking branch 'origin/develop' into fb_long_is_the_new_int

labkey-matthewb committed:
asInteger()

labkey-matthewb committed:
remove (Integer) casts

labkey-matthewb committed:
remove (Integer) casts
bulk search/replace

labkey-matthewb committed:
Merge remote-tracking branch 'origin/develop' into fb_long_is_the_new_int

labkey-matthewb committed:
convert()

labkey-matthewb committed:
remove (Integer) casts

labkey-matthewb committed:
wrong branch

labkey-adam committed:
Update all ObjectIds to BIGINT

labkey-matthewb committed:
convert lineage -> Long

labkey-adam commented on api/src/org/labkey/api/exp/api/ExpLineageEdge.java:
```diff
@@ -1,67 +1,59 @@
     {
-        // TODO BIGINT conversion
         _fromObjectId = (int)fromObjectId;
         _toObjectId = (int)toObjectId;
```
@labkey-matthewb get rid of these casts?

labkey-matthewb committed:
CAST(objectid AS BIGINT)

labkey-matthewb committed:
I like BIGINT

labkey-matthewb committed:
int ain't it

labkey-adam commented on api/src/org/labkey/api/data/Table.java:
```diff
@@ -913,7 +913,7 @@ else if (column.isAutoIncrement())
                     _setProperty(returnObject, autoIncColumn.getName(), rs.getLong(1));
                 else
-                    _setProperty(returnObject, autoIncColumn.getName(), rs.getInt(1));
+                    _setProperty(returnObject, autoIncColumn.getName(), rs.getLong(1));
```
Should make this unconditional

labkey-matthewb committed:
StatementDataIterator shouldn't use getRowError() since it batches and the input may not be on a row.

labkey-matthewb committed:
Merge remote-tracking branch 'origin/develop' into fb_long_is_the_new_int

labkey-matthewb committed:
(Integer)

labkey-adam committed:
Make provenance bigint-aware

labkey-matthewb committed:
(Integer)

labkey-matthewb committed:
(Integer) bulk replace

labkey-matthewb committed:
(Integer)

labkey-matthewb committed:
LongHashSet

labkey-matthewb committed:
equalsTo()

labkey-matthewb committed:
mixed Map<> fix

labkey-matthewb committed:
mixed Map<> fix

labkey-matthewb committed:
equalsTo()
Not sure which test hit this.

labkey-matthewb committed:
LongArrayList

labkey-matthewb committed:
ReferenceBlot and DataState fixes

labkey-adam committed:
Log the stack trace

labkey-adam committed:
Merge remote-tracking branch 'origin/fb_long_is_the_new_int' into fb_long_is_the_new_int

labkey-matthewb committed:
Sample status

labkey-matthewb committed:
LongHashSet

labkey-matthewb committed:
LongHashSet structural search/replace

labkey-matthewb committed:
Object.equals()

labkey-adam committed:
Fix assay QC state updates?

labkey-adam committed:
Merge remote-tracking branch 'origin/fb_long_is_the_new_int' into fb_long_is_the_new_int

labkey-adam committed:
Merge from develop

labkey-matthewb committed:
remove unnecessary if/else in Table.insert()

labkey-matthewb committed:
instanceof Integer
probably need a new helpers like
isIntegerObject() // instanceof Integer or Long
asIntegerOrNull() // null if not isIntegerObject()
asLongOrNull()    // null if not isIntegerObject()

labkey-adam commented on api/src/org/labkey/api/data/BaseColumnInfo.java:
```diff
@@ -2092,6 +2092,13 @@ public int getIntValue(ResultSet rs) throws SQLException
+    @Override
+    public long getLongValue(ResultSet rs) throws SQLException
+    {
+        // UNDONE
```
```suggestion
```

labkey-adam committed:
Update api/src/org/labkey/api/data/BaseColumnInfo.java

labkey-adam commented on api/src/org/labkey/api/dataiterator/SimpleTranslator.java:
```diff
@@ -1597,7 +1607,15 @@ private int addBuiltinColumn(SpecialColumn e, boolean allowPassThrough, TableInf
+
+    public int addSequenceColumn(ColumnInfo col, Container sequenceContainer, String sequenceName, @Nullable Long sequenceId, @Nullable Integer batchSize, @Nullable Long minValue)
+    {
+        // TODO BIGITN
```
```suggestion
        // TODO BIGINT
```

labkey-adam committed:
Update api/src/org/labkey/api/dataiterator/SimpleTranslator.java

labkey-adam commented on experiment/src/org/labkey/experiment/api/SampleTypeUpdateServiceDI.java:
```diff
@@ -413,8 +417,6 @@ public int loadRows(User user, Container container, DataIteratorBuilder rows, Da
             if (rows instanceof DataLoader dataLoader) // junit test uses ListofMapsDataIterator
             {
-                if (!context.isCrossTypeImport() && dataLoader.getColumnInfoMap().isEmpty())
-                    dataLoader.setKnownColumns(getQueryTable().getColumns());
```
Please undo for this PR

labkey-matthewb committed:
remove (Integer) casts

labkey-matthewb committed:
Merge branch 'fb_long_is_the_new_int' of https://github.com/LabKey/platform into fb_long_is_the_new_int

labkey-matthewb committed:
Merge remote-tracking branch 'origin/develop' into fb_long_is_the_new_int
